src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs

```diff
@@ -86,24 +89,26 @@ private XmlValueSerialization BuildXmlValueSerialization(Schema schema, bool isN
                     BuilderHelpers.GetSerializationFormat(schema));
         }
 
-        private JsonSerialization BuildSerialization(Schema schema, bool isNullable)
+        private JsonSerialization BuildSerialization(Schema schema, bool isNullable, VirtualParameter[] flattenedParameters)
         {
             switch (schema)
             {
                 case ConstantSchema constantSchema:
-                    return BuildSerialization(constantSchema.ValueType, constantSchema.Value.Value == null);
+                    return BuildSerialization(constantSchema.ValueType, constantSchema.Value.Value == null, Array.Empty<VirtualParameter>());
                 case ArraySchema arraySchema:
                     return new JsonArraySerialization(
                         _typeFactory.CreateType(arraySchema, isNullable),
-                        BuildSerialization(arraySchema.ElementType, false),
+                        BuildSerialization(arraySchema.ElementType, false, Array.Empty<VirtualParameter>()),
                         _typeFactory.CreateImplementationType(arraySchema, isNullable));
                 case DictionarySchema dictionarySchema:
                     return new JsonObjectSerialization(
                         _typeFactory.CreateType(dictionarySchema, isNullable),
                         Array.Empty<JsonPropertySerialization>(),
-                        new JsonDynamicPropertiesSerialization(BuildSerialization(dictionarySchema.ElementType, false)),
+                        new JsonDynamicPropertiesSerialization(BuildSerialization(dictionarySchema.ElementType, false, Array.Empty<VirtualParameter>())),
                         _typeFactory.CreateImplementationType(dictionarySchema, isNullable)
                         );
+                case ObjectSchema objectSchema when flattenedParameters.Any():
+                    return BuildJsonObjectSerialization(objectSchema, (ObjectType)_library.FindTypeForSchema(objectSchema), flattenedParameters);
```

Comment:
Why don't you pass the ObjectType that we got into the parent method here?

Reply:
Hm, shouldn't we have this logic in BuildObject?

Reply:
There is no ObjectType from the parent method.

Reply:
Why not build object?

Reply:
Because BuildSerialization is specifically for JSON (the method this code is in). BuildObject is to determine which media type builder to call. Here, we know which to call because we're in JSON.

Reply:
Why is it specific to JSON?

Reply:
We don't have model flattening for XML (https://github.com/Azure/autorest.csharp/issues/433), so a similar method won't work until then.

This place in the code is specifically a JSON codepath. I'll update the name to be more appropriate, so it is clear. You can see usage here:
![image](https://user-images.githubusercontent.com/17788297/77091501-e209fa80-69c5-11ea-8568-5f5ce2ed8f53.png)

Reply:
Then make https://github.com/Azure/autorest.csharp/blob/8e78e6f326f361dacde8e70c3396e74601b34697/src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs#L208 public and call it directly.

Reply:
I don't like this roundabout way of getting `ObjectType` when most methods just take them as parameter.

Reply:
I'm so confused. We are in the class that contains `BuildJsonObjectSerialization`, so making it public has no effect. The parent calls are to the method `Build`, which is public, and doesn't take `ObjectType`. `Build` itself has no association to ObjectType, because you're normally not building an object. `BuildObject` is for building objects. However, we're now utilizing the object building mechanism to build it in-line. That's the trade-off of doing this style of outputting. This section of code is for outputting values, and values normally have methods to call to do their serialization. Here, we're leveraging the object serialization in the value serialization pipeline, which is exactly what we agreed upon. Make sense?